### DISCUSSION TITLE
Does scheduler mandatory for control job number?

### DISCUSSION THREAD

**Discussion by pee8379** [deepmodeling/dpgen#739] [Q&A] [answered]
I am trying to running dpgen in my local desktop machine. (one 12-core cpu, 2 gpu)

As far I figured out, number of task in each job is controlled by "group_size" in machine json file. 
However, I keep failing to limit number of job submitted to run. 
It is predictable during "train", since "numb_models" is number of process submitted, but during "model_devi", hundreds of new processes try to run in same time, breaks entire system and make practically impossible to run dpgen in my desktop.

I try large "group_size" after read "Note that group_size should be set as large as possible to ensure there is only one job and avoid running multiple jobs at the same time." In [doc/examples/shell.md](https://github.com/deepmodeling/dpdispatcher/blob/master/doc/examples/shell.md) , but it didn't help.

I found some local shell machine examples in this github, (discussions #666 , [machine-local-4GPU.json](https://github.com/deepmodeling/dpgen/blob/master/examples/machine/DeePMD-kit-1.0/machine-local-4GPU.json) , [machine-local.json](https://github.com/deepmodeling/dpgen/blob/master/examples/machine/DeePMD-kit-1.x/machine-local.json) ) , but all use old dispatcher style. 
Does using old dispatcher will solve this problem?

If controlling number of running processes is impossible without scheduler, I considering to learning how to use slurm. This might be most common, easy option, right?

Thanks.

**Comment by AnguseZhang** [comment, marked as answer]
I think that setting`group_size` to be  a large number can solve your problem. It's probable that there's wrong with your `machine.json`. Could you post your `machine.json`?

**Reply by pee8379** [reply]
[machine_local_shell_pwscf.txt](https://github.com/deepmodeling/dpgen/files/8791014/machine_local_shell_pwscf.txt)

Yes, you are right. large group_size can was solution. Mixing old and new dpdispatcher style confuses me.

However, It seems number of POSCAR in sys_configs seems also affect number of submitting jobs. 
Is there a equation or other methods to predict number of maximum processes using group_size?

Anyway, I upload working machine.json file for local, single node, shell, pwscf, with newest dpgen/deepmd/dpdispatcher releases. (0.10.5, 2.1.1, 0.4.9)
I hope this help others who need example for this specific configuration.

**Comment by njzjz** [comment]
Added support to infinite `group_size` in https://github.com/deepmodeling/dpdispatcher/pull/194.
